### PR TITLE
[node-telegram-bot-api] typings for telegram bot api v6.5

### PR DESCRIPTION
**Pull request by yokkkoso**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [ ] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://core.telegram.org/bots/api#february-3-2023
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

File: types/node-telegram-bot-api/index.d.ts [hunk 1]
```diff
@@ -299,17 +303,13 @@ declare namespace TelegramBot {
...
     interface RestrictChatMemberOptions {
         until_date?: number | undefined;
-        can_send_messages?: boolean | undefined;
-        can_send_media_messages?: boolean | undefined;
-        can_send_polls?: boolean | undefined;
-        can_send_other_messages?: boolean | undefined;
-        can_add_web_page_previews?: boolean | undefined;
-        can_change_info?: boolean | undefined;
-        can_invite_users?: boolean | undefined;
-        can_pin_messages?: boolean | undefined;
```
Comment (XieJiSS) [inline types/node-telegram-bot-api/index.d.ts:309]: Are these options still available if use_independent_chat_permissions is set to true?

Comment (yokkkoso) [reply types/node-telegram-bot-api/index.d.ts:309]: as said [here](https://core.telegram.org/bots/api#restrictchatmember) if `use_independent_chat_permissions` true: some permissions doesn't inherit from other permissions so we can use all this permissions only when `use_independent_chat_permissions` seted to true

Comment (XieJiSS) [reply types/node-telegram-bot-api/index.d.ts:309]: Seems like the telegram bot api has introduced a beaking change here. Anyway this part LGTM now

**Review by XieJiSS — APPROVED** [review]

**Comment by theimo1221** [comment]
This introduces TS2300 Errors which are resolved in #64402

**Pull request merged**